scripts/parser/backend/cmakecachestring.py

```diff
@@ -0,0 +1,41 @@
+import re
+
+class cmakecache_string(str):
+   """
+   simple utility string that abstracts away some details from the parsing
+   process
+   """
+   regex_expression =  r"^([A-Za-z_0-9]+?):([A-Za-z]+?)=(.*)$"
+
+   def user_comment(self) -> str:
+      """
+      if a line starts with // we assume it's a comment, and we strip away the
+      newline that is added by CMake
+      """
+      return self[2:].rstrip()
+
+   def is_comment(self) -> bool:
+      return self[:2] == "//"
```

Comment:
OK, but what about the comments starting with '#'? How do you skip them?

Reply:
Currently using, 
 ```python
  return self[:2] == "//" or self[0] == "#"
```
should i add a unit test for it?

Reply:
Sure, why not.

Reply:
Done.